classes/Image.php

```diff
@@ -941,17 +946,21 @@ public function getPathForCreation()
      */
     private function deleteAutoGeneratedImage(array $imageType, string $imageFormat, array $filesToDelete): array
     {
-        $configuration = SymfonyContainer::getInstance()->get('prestashop.adapter.legacy.configuration');
-
         // Regular thumbnail
         $filesToDelete[] = $this->image_dir . $this->getExistingImgPath() . '-' . $imageType['name'] . '.' . $imageFormat;
 
         // Old 2x thumbnail, if present
         $filesToDelete[] = $this->image_dir . $this->getExistingImgPath() . '-' . $imageType['name'] . '2x.' . $imageFormat;
 
-        // Watermarked thumbnail, if present
-        if ($configuration->get('WATERMARK_HASH')) {
-            $filesToDelete[] = $this->image_dir . $this->getExistingImgPath() . '-' . $imageType['name'] . '-' . $configuration->get('WATERMARK_HASH') . '.' . $imageFormat;
+        $sfContainer = SymfonyContainer::getInstance();
+
+        if ($sfContainer instanceof ContainerInterface) {
+	        $configuration = $sfContainer->get('prestashop.adapter.legacy.configuration');
```

Comment:
We are avoid this string to get service from the container.
For `prestashop.adapter.legacy.configuration`, you can use instead:
```suggestion
	        $configuration = $sfContainer->get(Configuration::class);
```
Don't forget to add `use PrestaShop\PrestaShop\Adapter\Configuration` at the top of this file.